docs/LocalesMenuButton.md

```diff
@@ -11,38 +11,54 @@ The `<LocalesMenuButton>` component, also known as the "language switcher", disp
 
 ## Usage
 
-Add the `<LocalesMenuButton>` to a custom `<AppBar>`:
+**Tip**: For most users, this component will be automatically added to `<AppBar>` if the `i18nProvider` is configured properly to return a list of available locales. React-admin will use the optional `getLocales` method of your `i18nProvider` (or the `availableLocales` parameter if you are using `polyglotI18nProvider`) to generate a list of locale menu items for this component.
+
+For advanced users who wish to add `<LocalesMenuButton>` elsewhere on the `<AppBar>` or to a custom configuation page, they can do the following:
+
+Define an `i18nProvider` with `getLocales` omitted. (or `polyglotI18nProvider` with `availableLocales` omitted).
```

Comment:
If we want to put the `<LocalesMenuButton>` in a custom place, the button cannot take the result from getLocales in i18nProvider. 
See: https://github.com/marmelab/react-admin/blob/47d43077f70d32469df9badb53ee1f7fdbe42e2c/packages/ra-ui-materialui/src/layout/AppBar.tsx#L97 if there are more than 1 locales from getLocales, this button will automatically show up in `<AppBar>` which breaks the purpose to put it in a custom place.

Hence, we have to pass the languages prop manually to it.

Reply:
I don't fully agree. To me if you do not want the `<LocalesMenuButton />` in your `<AppBar>`, then you should provide a custom `<AppBar>`.

Not passing the locales to the i18nProvider only to disable the `<LocalesMenuButton />` in the `<AppBar>` does not seem like the way to go imo.

Reply:
Change the custom AppBar to use MUI AppBar component instead of React Admin AppBar component.

Apply review feedback so i18nProvider should still be able to accept a list of locales in this example.